completion_aggregator/migrations/0004_index_stalecompletion.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.8 on 2018-10-18 00:13
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.utils.timezone
+import model_utils.fields
+import opaque_keys.edx.django.models
+import six
+
+
+BATCH_SIZE = 1000
+
+
+def copy_data(apps, schema_editor):
+    """
+    Add indexes to completion_aggregator_stalecompletion without locking.
+
+    Copy data into the new stalecompletionnew table in batches, then rename
+    both tables, then do a final copy of any new data that came in.
+    """
+
+    copy_sql_tmpl = """
+        INSERT INTO %(target)s (`created`, `modified`, `username`, `course_key`, `block_key`, `force`, `resolved`)
+            SELECT `created`, `modified`, `username`, `course_key`, `block_key`, `force`, `resolved`
+                FROM %(source)s
+                WHERE id > %%s
+                ORDER BY id
+                LIMIT %%s OFFSET %%s
+    """
+    StaleCompletion = apps.get_model("completion_aggregator", "StaleCompletion")
+    cursor = schema_editor.connection.cursor()
+    max_id = StaleCompletion.objects.order_by('-id')[0].id
+    count = StaleCompletion.objects.count()
+    copy_sql = copy_sql_tmpl % {
+        'source': 'completion_aggregator_stalecompletion',
+        'target': 'completion_aggregator_stalecompletionnew',
+    }
+    for offset in six.moves.range(0, count, BATCH_SIZE):
+        cursor.execute(copy_sql, [0, BATCH_SIZE, offset])
+
+    cursor.execute(
+        """
+        RENAME TABLE
+            completion_aggregator_stalecompletion TO completion_aggregator_stalecompletionold,
+            completion_aggregator_stalecompletionnew TO completion_aggregator_stalecompletion
+        """
+    )
+    end_count = StaleCompletion.objects.filter(id__gt=max_id).count()
```

Comment:
After renaming tables, wouldn't this look inside the table that we just created, which would technically have no ID greater than `max_id` (which could still happen from what I mentioned in the comment above, but even then it'd be missing some things)?